### PR TITLE
feat: detect config drift between user config and template

### LINKED ISSUE #1041: Detect missing config keys by comparing instance/config.yaml against template
## Summary

On startup, compare the user's `instance/config.yaml` against `instance.example/config.yaml` (the template) and warn about keys present in the template but missing from the live config. This helps users discover new features they may not know about.

## Context

- `config_validator.py` already validates types and detects typos, but works from a hardcoded schema — it doesn't know about keys the user simply never added.
- `startup_manager.py` already calls `validate_config()` at boot, so the hook point exists.
- `diagnostics/config_check.py` (part of #746 `/doctor` infra) could also host this check.
- A recent manual comparison found 2 high-priority gaps and 8+ optional features documented in the template but absent from a real config.

## Proposed behavior

1. At startup, parse both `instance.example/config.yaml` and `instance/config.yaml`
2. Recursively diff the key trees
3. Log missing keys as **advisory** warnings (not errors) — the agent should still start normally
4. Example output:
   ```
   ℹ️  Config drift: 3 keys in template not in your config.yaml:
       - max_open_prs (under: tools)
       - skill_timeout (top-level)
       - auto_update.notify (under: auto_update)
     → See instance.example/config.yaml for documentation
   ```

## Notes

- Advisory only — missing keys are not errors, just opportunities
- Should ignore comments and only compare YAML keys
- Could also surface in `/doctor` (#746) and `/status`

### PR DESCRIPTION
**Pull request by sukria-koan0**
## What
Compares `instance/config.yaml` against `instance.example/config.yaml` at startup to report keys present in the template but missing from the user's config.

## Why
Closes #1041. After updates, users may not know about new config features. This surfaces them as advisory info — no errors, no blocks, just "hey, these exist."

## How
- `detect_config_drift(koan_root, user_config)` recursively collects key trees from both configs, diffs them, and filters out child keys when the parent section is already reported missing (e.g., reports `auto_update` not `auto_update.enabled` + `auto_update.notify` + `auto_update.check_interval`)
- Integrated into `validate_and_warn()` (startup path) via new `koan_root` parameter
- Also integrated into `/doctor` diagnostics (`config_check.py`) with severity `info`
- Logs at `info` level, never `warn` — this is advisory, not an error

## Testing
- 17 new tests covering: identical configs, missing top-level keys, missing nested sections, parent filtering, file loading, edge cases (empty template, missing files), real-world scenario
- 60 total tests in `test_config_validator.py`, 9383 total suite green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
### Quality Report

**Changes**: 4 files changed, 274 insertions(+), 6 deletions(-)

**Code scan**: clean

**Tests**: passed (10 PASSED)

**Branch hygiene**: clean

*Generated by Kōan post-mission quality pipeline*

### CODE REVIEW

File: koan/tests/test_config_validator.py [hunk 1]
```diff
@@ -330,3 +330,156 @@ def test_no_warnings_no_output(self, capsys):
...
+
+    def test_drift_detection_with_koan_root(self, tmp_path, capsys):
+        """validate_and_warn with koan_root triggers drift detection."""
+        import yaml
+
+        template = {"max_runs_per_day": 20, "auto_update": {"enabled": True}}
+        user = {"max_runs_per_day": 20}
+
+        (tmp_path / "instance.example").mkdir()
+        (tmp_path / "instance.example" / "config.yaml").write_text(yaml.dump(template))
```
Comment (atoomic) [inline koan/tests/test_config_validator.py]: looks like we are going to want on `default` values in the config file but commented

**Comment by sukria-koan0** [comment]
## Rebase: feat: detect config drift between user config and template

Branch `koan/config-drift-detection` rebased onto `main` and force-pushed.

**Diff**: 4 files changed, 352 insertions(+), 6 deletions(-)

Review feedback was analyzed and applied.

### Changes

 a summary of the changes:

**Summary of changes (for commit message):**

- **Skip commented-out keys in config drift detection** per reviewer request: keys that are commented out in the user's `instance/config.yaml` (e.g. `# debug: false`) are now excluded from the drift report. A commented key means the user is aware of it and chose to use the default value.
- Added `_find_commented_keys()` helper that scans raw config file text for lines matching `# key_name:` patterns and returns the set of commented key names.
- `detect_config_drift()` now reads the raw user config file to find commented keys, then filters them out of the missing-keys list alongside the existing parent-key filtering.
- Added 4 new tests: commented top-level key excluded, commented nested key excluded, fully commented section excluded, graceful handling when no user config file exists on disk.
- Updated `test_drift_detection_with_koan_root` to write a user config file on disk (needed for the comment-scanning path).

### Actions

- Rebased `koan/config-drift-detection` onto `origin/main`
- Applied review feedback
- Force-pushed `koan/config-drift-detection` to origin
- CI check enqueued (async)

### CI

CI will be checked asynchronously.

---
_Automated by Kōan_

**Comment by sukria** [comment]
Merged into main.
